cost/aws/object_storage_optimization/aws_object_storage_optimization.pt

```diff
@@ -178,9 +239,72 @@ datasource "ds_aws_list_s3_objects" do
+  parameters "ds_aws_list_s3_objects", "param_deep_archive_days", "param_glacier_days"
+  result "results"
+  code <<-EOS
+    results = []
+    var glacier_date = new Date();
+    var deep_archive_date = new Date();
+    var glacier_days_to_move = parseInt(param_glacier_days,10);
+    var deep_archive_days_to_move = parseInt(param_deep_archive_days,10);
+    glacier_date = glacier_date.setDate(glacier_date.getDate() - glacier_days_to_move);
+    deep_archive_date = deep_archive_date.setDate(deep_archive_date.getDate() - deep_archive_days_to_move);
+    glacier_date = (new Date(glacier_date)).setHours(23, 59, 59, 999);
+    deep_archive_date =(new Date(deep_archive_date)).setHours(23, 59, 59, 999);
+
+    _.each(ds_aws_list_s3_objects, function (s3Object) {
+      if (s3Object.storage_class !== "GLACIER" && s3Object.storage_class !== "DEEP_ARCHIVE"){
+        var last_modified_date = (new Date(s3Object["last_modified"])).setHours(23, 59, 59, 999);
+        var modify_storage_class_to="";
+        if(param_glacier_days.length > 0 && param_deep_archive_days.length == 0 &&  last_modified_date <= glacier_date){
+          //When param_deep_archive_days is blank.
+          modify_storage_class_to = "GLACIER";
+        }else if(param_glacier_days.length == 0 && param_deep_archive_days.length > 0 && last_modified_date <= deep_archive_date){
+          //When param_glacier_days is blank.
+          modify_storage_class_to = "DEEP_ARCHIVE";
+        }else if(param_glacier_days.length > 0 && param_deep_archive_days.length > 0 && (last_modified_date <= glacier_date || last_modified_date <= deep_archive_date)){
+          if(glacier_days_to_move < deep_archive_days_to_move){
+            if(last_modified_date <= glacier_date && last_modified_date >= deep_archive_date){
+              modify_storage_class_to = "GLACIER";
+            }else{
+              modify_storage_class_to = "DEEP_ARCHIVE";
+            }
+          }else if(glacier_days_to_move > deep_archive_days_to_move){
+            if(last_modified_date <= deep_archive_date && last_modified_date >= glacier_date){
+              modify_storage_class_to = "DEEP_ARCHIVE";
+            }else{
+              modify_storage_class_to = "GLACIER";
+            }
+          }
+        }
+        if(modify_storage_class_to.length > 0){
+          results.push({
+            "bucket_name": s3Object["bucket_name"],
+            "region":s3Object["region"],
+            "bucket_creation_date": s3Object["bucket_creation_date"],
+            "host": s3Object["host"],
+            "object_key": s3Object["object_key"],
+            "storage_class": s3Object["storage_class"],
+            "last_modified": s3Object["last_modified"],
+            "modify_storage_class_to": modify_storage_class_to,
+            "object_size": s3Object["object_size"]
+          });
+        }
+      }
+    });
+    results = _.sortBy(results, 'region');
+    console.log(results)
```

Comment:
console.log statements are used for debugging and should be removed from production policies